### DISCUSSION TITLE
Use of class in class results in 'name <name> isn't defined'

### DISCUSSION THREAD

**Discussion by pholat** [micropython/micropython#10479] [Using MicroPython] [answered]
Minimal viable code:
```
class Name():
    O = 0
    R =1
    L = 2


class Test():
    def __init__(self):
        pass

    def set(self, name):
        if name == Name.L:
            print(name)

if __name__ == "__main__":
    m = Test()
    m.set(Name.L)
```

results in `name Name isn't defined`

I started with  `class Name` defined in `Test`, that at least to my best python knowledge should also be permitted and it resulted in the same Error. Importing `Name` with `import` resulted in the same behavior. As I found, any use of class Name in class Test will fail miserably.
I would be grateful to pointing me out how I could succeed in using class in class.

**Comment by pholat** [comment, marked as answer]
After clean start I can't reproduce it :|

**Reply by mtravers** [reply]
Same thing happened to me, and a reboot fixed it. Weird.

**Reply by pholat** [reply]
Thanks for the confirmation, I felt ridiculous when I rebooted the board and it worked just fine.
